modules/features/manatime_feature_timesheets/manatime_feature_timesheets.module

```diff
@@ -59,3 +76,183 @@ function manatime_feature_timesheets_cron() {
+  return l(t('add row'), 'timesheet/add-new-form/' . $nid . '/nojs', array('attributes' => array('class' => 'ctools-use-modal')));
+}
+
+/**
+ * Helper function to populate the project dropdown.
+ */
+function timesheet_get_projects_options($timesheet) {
+  $list = array();
+  $query = new EntityFieldQuery();
+  $result = $query
+    ->entityCondition('entity_type', 'node')
+    ->entityCondition('bundle', 'projects')
+    ->execute();
+  if (isset($result['node'])) {
+    $nids = array_keys($result['node']);
+    $projects = node_load_multiple($nids);
+  }
+  foreach ($projects as $project) {
+    // Get only projects with tasks.
+    if (isset($project->field_tasks[LANGUAGE_NONE])) {
+      $list[$project->nid] = $project->title;
+    }
+  }
+  return $list;
+}
+
+/**
+ * Helper function to populate the task dropdown.
+ */
+function timesheet_get_tasks_options($selected, $timesheet) {
+  $list = array();
+  $nids = array();
+  $project = node_load($selected);
+  // Get the ids of tasks.
+  foreach ($project->field_tasks[LANGUAGE_NONE] as $nid) {
+    $nids[] = $nid['target_id'];
+  }
+  $tasks = node_load_multiple($nids);
+  foreach ($tasks as $task) {
+    // Get only the tasks not recorded in the reports.
+    if (empty($timesheet->getTimeEntriesPerTask($record['target_id']))) {
+      $list[$task->nid] = $task->title;
+    }
+  }
+  return $list;
+}
+
+/**
+ * Ajax menu callback.
+ */
+function manatime_feature_timesheets_add_row_callback($nid, $ajax) {
+  if ($ajax) {
+    ctools_include('ajax');
+    ctools_include('modal');
+
+    $form_state = array(
+      'ajax' => TRUE,
+      'title' => t('Adding'),
+    );
+
+    $form_state['build_info']['args'] = array($nid);
+
+    // Use ctools to generate ajax instructions for the browser to create
+    // a form in a modal popup.
+    $output = ctools_modal_form_wrapper('manatime_feature_timesheets_add_row_form', $form_state);
+
+    // If the form has been submitted, there may be additional instructions
+    // such as dismissing the modal popup.
+    if (!empty($form_state['ajax_commands'])) {
+      $output = $form_state['ajax_commands'];
+    }
+
+    // Return the ajax instructions to the browser via ajax_render().
+    print ajax_render($output);
+    drupal_exit();
+  }
+  else {
+    return drupal_get_form('manatime_feature_timesheets_add_row_form');
+  }
+}
+
+/**
+ * Update the section to the dropdown task.
+ */
+function timesheet_dependent_dropdown_callback($form, $form_state) {
+  return $form['dropdown_task'];
+}
+
+/**
+ * Drupal form submit handler.
+ */
+function manatime_feature_timesheets_add_row_form_submit(&$form, &$form_state) {
+  // Generate the new link using the submitted text value.
+  $timesheet = new Timesheet($form_state['build_info']['args'][0]);
+  $timesheet->generateRecords();
+  $project = node_load($form_state['complete form']['dropdown_project']['#value']);
+  $task = node_load($form_state['complete form']['dropdown_task']['#value']);
```

Comment:
you're loading a node only to use title. Please use EntityMetadataWrapper instead. It allows you to have lazy loading and get a better performance.